### PR TITLE
add timezone migration toggle

### PR DESCRIPTION
**Pull request by dannyroberts**
- implement a way to get the current request regardless of context
  using threadlocals
- use this in timezone conversions instead of a single global variable

### CODE REVIEW

File: corehq/toggles.py [hunk 1]
```diff
@@ -366,3 +366,11 @@ def toggles_dict(username=None, domain=None):
...
     'Enable payment of invoices by bulk credit payments and invoice generation for wire tranfers',
 )
+
+USE_NEW_TIMEZONE_BEHAVIOR = StaticToggle(
+    'use_new_timezone_behavior',
+    ("Enable properly dealing with phone timestamps "
+     "during submission and in reports. "
+     "(Please do not set manually, "
+     "because it has to be accompanied by a migration.)")
+)
```
Comment (snopoke) [inline corehq/toggles.py]: Given how you're using this below you need to set the namespaces here to `[NAMESPACE_DOMAIN]`

Comment (dannyroberts) [inline corehq/toggles.py]: Thanks

**Comment by dannyroberts** [comment]
@czue I think this is ready to go. Tested it on staging and the threading.local implemention of `get_request()` seems to be working. Also this stackoverflow answer http://stackoverflow.com/a/15594647/240553 and the gevent documentation confirm that when you've gevent monkey patched all, thread locals actually become local to individual greenlets instead, which is exactly what we want

@NoahCarnahan buddy fyi

**Comment by NoahCarnahan** [comment]
Looks good to me. I'll let czue or whoever merge as he probably has more context.

File: corehq/util/timezones/conversions.py [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+    """
+    The timezone data migration happening some time in Apr-May 2015
+    will shift all phone times (form.timeEnd, case.modified_on, etc.) to UTC
+    so functions that deal with converting to or from phone times
+    use this function to decide what type of timezone conversion is necessary
+
+    """
+    _default = False
+    request = get_request()
+    if not request or not request.domain:
```
Comment (czue) [inline corehq/util/timezones/conversions.py]: will this fail hard if request doesn't have a domain attribute?

File: corehq/util/view_utils.py [hunk 3]
```diff
@@ -103,20 +104,7 @@ def inner(request, *args, **kwargs):
...
-    frame = None
-    try:
-        for f in inspect.stack()[1:]:
-            frame = f[0]
-            code = frame.f_code
-            if code.co_varnames and code.co_varnames[0] in ("request", "req"):
-                return frame.f_locals[code.co_varnames[0]]
-    finally:
-        del frame
+    return global_request.get_request()
```
Comment (czue) [inline corehq/util/view_utils.py]: nice

File: corehq/util/timezones/conversions.py (hunk 2 above)
Comment (dannyroberts) [inline corehq/util/timezones/conversions.py]: Good point, fixed

**Comment by dannyroberts** [comment]
@czue this should be good to go now

**Pull request merged**